write-fonts/src/tables/glyf.rs

```diff
@@ -0,0 +1,298 @@
+    TooSmall,
+    MissingMove,
+}
+
+/// A helper trait for converting other point types to open-type compatible reprs
+pub trait OtPoint {
+    fn get(self) -> (i16, i16);
+}
+
+impl OtPoint for kurbo::Point {
+    fn get(self) -> (i16, i16) {
+        (ot_round(self.x as f32), ot_round(self.y as f32))
+    }
+}
+
+impl OtPoint for (i16, i16) {
+    fn get(self) -> (i16, i16) {
+        self
+    }
+}
+
+// adapted from simon:
+// https://github.com/simoncozens/rust-font-tools/blob/105436d3a617ddbebd25f790b041ff506bd90d44/otmath/src/lib.rs#L17
+fn ot_round(val: f32) -> i16 {
+    (val + 0.5).floor() as i16
+}
+
+impl Contour {
+    /// Create a new contour begining at the provided point
+    pub fn new(pt: impl OtPoint) -> Self {
+        let (x, y) = pt.get();
+        Self(vec![CurvePoint::on_curve(x, y)])
+    }
+
+    /// The total number of points in this contour
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// `true` if this contour is empty
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    /// Add a line segment
+    pub fn line_to(&mut self, pt: impl OtPoint) {
+        let (x, y) = pt.get();
+        self.0.push(CurvePoint::on_curve(x, y));
+    }
+
+    /// Add a quadratic curve segment
+    pub fn curve_to(&mut self, p0: impl OtPoint, p1: impl OtPoint) {
+        let (x0, y0) = p0.get();
+        let (x1, y1) = p1.get();
+        self.0.push(CurvePoint::off_curve(x0, y0));
+        self.0.push(CurvePoint::on_curve(x1, y1));
+    }
+}
+
+impl SimpleGlyf {
+    /// Attempt to create a simple glyph from a kurbo `BezPath`
+    ///
+    /// The path may contain only line and quadratic bezier segments. The caller
+    /// is responsible for converting any cubic segments to quadratics before
+    /// calling.
+    ///
+    /// Returns an error if the input path is malformed; that is, if it is empty,
+    /// contains cubic segments, or does not begin with a 'move' instruction.
+    //TODO: figure out a more general API? maybe a nested builder thing, where you
+    //build contours, and from those contours build a glyph? idk?
+    pub fn from_kurbo(path: &BezPath) -> Result<Self, BadKurbo> {
+        let mut contours = Vec::new();
+        let mut current = None;
+
+        for el in path.elements() {
+            match el {
+                kurbo::PathEl::MoveTo(pt) => {
+                    if let Some(prev) = current.take() {
+                        contours.push(prev);
+                    }
+                    current = Some(Contour::new(*pt));
+                }
+                kurbo::PathEl::LineTo(pt) => {
+                    current.as_mut().ok_or(BadKurbo::MissingMove)?.line_to(*pt)
+                }
+                kurbo::PathEl::QuadTo(p0, p1) => current
+                    .as_mut()
+                    .ok_or(BadKurbo::MissingMove)?
+                    .curve_to(*p0, *p1),
+                kurbo::PathEl::CurveTo(_, _, _) => return Err(BadKurbo::HasCubic),
```

Comment:
We could lower to quads here. I’m not sure what the implications might be.

Reply:
yea, I wanted to leave the conversion out of it since it might get complicated with variable fonts; the premise here is that you've already done your conversion and verified that it's sane etc.

Reply:
+1 to leave it to higher levels. For example I would expect the compiler to  look at every instance of a contour and safely lower it while retaining interpolation compatibility.